internal/jsonutil/graphql.go

```diff
@@ -224,10 +250,28 @@ func (d *decoder) decode() error {
 					if v.Kind() != reflect.Slice {
 						continue
 					}
-					v.Set(reflect.MakeSlice(v.Type(), 0, 0)) // v = make(T, 0, 0).
+					newSlice := reflect.MakeSlice(v.Type(), 0, 0) // v = make(T, 0, 0).
+					switch v.Len() {
+					case 0:
+						// if there is no template we need to create one so that we can
+						// handle both cases (with or without a template) in the same way
+						newSlice = reflect.Append(newSlice, reflect.Zero(v.Type().Elem()))
+					case 1:
+						// if there is a template, we need to keep it at index 0
+						newSlice = reflect.Append(newSlice, v.Index(0))
+					case 2:
+						msg := fmt.Sprintf("template slice can only have 1 item")
+						panic(msg)
```

Comment:
should we return error, and let users decide whether to panic themselves?

Reply:
sure